File: src/main/java/guru/xgm/language/angelscript/formatting/block/AngelScriptBlockSwitchStat.java
package guru.xgm.language.angelscript.formatting.block;

import com.intellij.formatting.Alignment;
import com.intellij.formatting.Block;
import com.intellij.formatting.Indent;
import com.intellij.lang.ASTNode;
import guru.xgm.language.angelscript.formatting.block.utils.AngelScriptBlockSettings;
import org.jetbrains.annotations.NotNull;

import static com.intellij.psi.formatter.FormatterUtil.isOneOf;
import static guru.xgm.language.angelscript.psi.AngelScriptTypes.CASE_STMT;

public class AngelScriptBlockSwitchStat extends AngelScriptBlockStat {
    public AngelScriptBlockSwitchStat(ASTNode myNode, Alignment myAlignment, Indent myIndent, AngelScriptBlockSettings settings, int braceStyle) {
        super(myNode, myAlignment, myIndent, settings, braceStyle);
    }

    @Override
    public Block makeSubBlock(@NotNull ASTNode childNode, Indent indent) {
        if (isOneOf(childNode, CASE_STMT))
            return new AngelScriptBlockCase(childNode, null, null, settings);

        return super.makeSubBlock(childNode, indent);
    }
}

File: src/main/kotlin/guru/xgm/language/angelscript/formatting/block/AngelScriptBlockSwitchStat.kt
package guru.xgm.language.angelscript.formatting.block

import com.intellij.formatting.Alignment
import com.intellij.formatting.Block
import com.intellij.formatting.Indent
import com.intellij.lang.ASTNode
import com.intellij.psi.formatter.FormatterUtil
import guru.xgm.language.angelscript.formatting.block.utils.AngelScriptBlockSettings
import guru.xgm.language.angelscript.psi.AngelScriptTypes

class AngelScriptBlockSwitchStat(
    myNode: ASTNode,
    myAlignment: Alignment?,
    myIndent: Indent?,
    settings: AngelScriptBlockSettings,
    braceStyle: Int
) : AngelScriptBlockStat(myNode, myAlignment, myIndent, settings, braceStyle) {
    override fun makeSubBlock(childNode: ASTNode, indent: Indent): Block {
        if (FormatterUtil.isOneOf(childNode, AngelScriptTypes.CASE_STMT)) return AngelScriptBlockCase(
            childNode,
            null,
            null,
            settings
        )

        return super.makeSubBlock(childNode, indent)
    }
}
